### DISCUSSION TITLE
Lost my entire timeline?

### DISCUSSION THREAD

**Discussion by tryinganon** [dialogic-godot/dialogic#2060] [Q&A]
Hey, so I'm new to both Godot and dialogic. Was working on a timeline and spent a TON of time on it, when suddenly it just... Disappears. I was trying to fix an error with a choice where only the first choice was showing up, moved something and everything was just gone. The timeline file is 0KB so I know it's just been wiped. So... Uh... How do I make sure this doesn't happen again?

Gonna have to start over on this entire project basically and don't want it to happen again.

Edit: Got 103 parse errors after a refresh and am now having to start over the ENTIRE project??

**Comment by CakeVR** [comment]
Hello, what version of Dialogic have you been using? If this is Dialogic 2, which Alpha?

If you encounter parse errors, your cache may have been invalidated. You can try to refresh the files in question or close Godot, delete the `import` folder in `.godot`, and restart.

I recommend to always use Git and commit often.

**Reply by tryinganon** [reply]
I was using alpha 12 when this happened, in hindsight using alpha builds of things I should really be more careful. When I had to restart everything, I switched to alpha 13, but I'll probably be forced back to alpha 12 because setting a variable to a random number isn't working lol (I'm debugging the crap out of it but I just cannot figure out why pick_random() in the Dialogic script is returning null lmao)

**Reply by CakeVR** [reply]
Make sure you have properly updated the Autoload path as described in the changelog of Dialogic 2 Alpha 13.
It should use the `Core` folder now instead of `Other`. We moved the handler around. This gets automatically upgraded, when you updated with the built-in updater.

**Reply by tryinganon** [reply]
I don't think that's it, since I ended up completely deleting Alpha 12 and my original game project (since 99% of the work was wiped anyway) and just starting completely from scratch, but with Alpha 13 instead of Alpha 12

Good to know there's a built-in updater though because I couldn't actually find that haha

Edit: Recreated the random returning null thing super easily outside of my project, so I know it's an issue with Alpha 13 (and that 12 didn't have it). New project, make a variable of either a float or an integer (default 0), make a timeline. In the timeline, have it set the variable to itself + random. Play it and get the error "Dialogic: Set Variable event failed because one value wasn't a float! [0, < null >]"
It does the same thing with all of them, not just when adding to itself, but having it set to requires breakpoints that I already disabled and I forgot which lines it was lol. But yeah the random thing is spitting out null

If you can't recreate it I guess it's on my end and I'll have to restart everything again but considering I legitimately just did that I feel like it's not a problem on my end?

(Ik that's not what this thread is about sorry if this is derailing/something I shouldn't do, idk github etiquette very well yet, but I figured I'd give more info anyway)

**Reply by CakeVR** [reply]
Thanks, please open an issue on our GitHub for this :)

**Reply by tryinganon** [reply]
Will do!

Edit: Solved it myself but it's still there so you guys can fix it easily! Someone just forgot to type a number apparently lol
